contrib/platform/test/com/sun/jna/platform/win32/Shell32Test.java

```diff
@@ -131,16 +135,63 @@ public void testResizeDesktopFromTop() throws InterruptedException {
 
     }
 
-    public void testSHGetKnownFolderPath()
-    {
-        int flags = ShlObj.KNOWN_FOLDER_FLAG.NONE.getFlag();
-        PointerByReference outPath = new PointerByReference();
-        HANDLE token = null;
-        GUID guid = KnownFolders.FOLDERID_Fonts;
-        HRESULT hr = Shell32.INSTANCE.SHGetKnownFolderPath(guid, flags, token, outPath);
+	public void testSHGetKnownFolderPath() {
+		int flags = ShlObj.KNOWN_FOLDER_FLAG.NONE.getFlag();
+		PointerByReference outPath = new PointerByReference();
+		HANDLE token = null;
+		GUID guid = KnownFolders.FOLDERID_Fonts;
+		HRESULT hr = Shell32.INSTANCE.SHGetKnownFolderPath(guid, flags, token, outPath);
 
-        Ole32.INSTANCE.CoTaskMemFree(outPath.getValue());
+		Ole32.INSTANCE.CoTaskMemFree(outPath.getValue());
 
-        assertTrue(W32Errors.SUCCEEDED(hr.intValue()));
+		assertTrue(W32Errors.SUCCEEDED(hr.intValue()));
+	}
+    
+	public void testSHEmptyRecycleBin() {
+		try {
+			File f = createTempFile();
+			W32FileUtils.getInstance().moveToTrash(new File[] { f });
+			int result = Shell32.INSTANCE.SHEmptyRecycleBin(null, null,
+					Shell32.SHERB_NOCONFIRMATION | Shell32.SHERB_NOPROGRESSUI | Shell32.SHERB_NOSOUND);
+			// for reasons I can not find documented, the function returns the following:
+			// -2147418113 when the recycle bin has no items in it
+			// 0 when the recycle bin has something in it and this call deletes
+			// it.
+			assertTrue(result == 0);
```

Comment:
```java
assertEquals("Unexpected failure code", ERROR_SUCCESS, result);
```
Try to provide as much information about the failure - in this case, not only that it has failed but also why.